packages/container/lib/container.js

```diff
@@ -153,18 +159,29 @@ Container.prototype = {
     assert('fullName must be a proper full name', this.registry.validateFullName(normalizedName));
 
     if (options.source) {
-      normalizedName = this.registry.expandLocalLookup(fullName, options);
+      let expandedFullName = this.registry.expandLocalLookup(fullName, options);
       // if expandLocalLookup returns falsey, we do not support local lookup
-      if (!normalizedName) {
-        return;
+      if (!EMBER_MODULE_UNIFICATION) {
+        if (!expandedFullName) {
+          return;
+        }
+
+        normalizedName = expandedFullName;
+      } else if (expandedFullName) {
+        // with ember-module-unification, if expandLocalLookup returns something,
+        // pass it to the resolve without the source
+        normalizedName = expandedFullName;
+        options = copy(options, true);
+        delete options.source;
```

Comment:
@iezer Right, `options` as an argument to `lookup` is only a contract with the `lookup` method. We don't document the argument as `resolverOptions` where it is implied the object is passed transparently.

A patch that build the object up by iterating `Object.keys` but skipping `source` would be ok, but really IMO just building the new object (in this case passing an empty object) is fine. Building a whole new object by iterating would be slow, calling the `delete` after copying is slow, and this is fairly performance sensitive code. Doing the fast and API-conservative thing is ok.